server.go

```diff
@@ -139,7 +139,8 @@ type Server struct {
 	quit               *grpcsync.Event
 	done               *grpcsync.Event
 	channelzRemoveOnce sync.Once
-	serveWG            sync.WaitGroup // counts active Serve goroutines for GracefulStop
+	serveWG            sync.WaitGroup // counts active Serve goroutines for Stop/GracefulStop
```

Comment:
Is there a possibility to use the same WG `serveWG`. 

atm, `serveWG.Done()` is called in the goroutine created by `Serve()` right after calling `s.handleRawConn()`. But instead could we defer the call to `serverWG.Done()` in the goroutine created by startStream func when calling `st.HandleStreams` (ie, the place where we are calling `handlersWg.Done()` in this change?

`Serve()` creates goroutines for every incoming connection and calls `s.handleRawConn()`. 
`s.handleRawConn()` spins off a goroutine calling `serveStreams()`. `serveStream()` calls `serverTransport.HandleStream()` which spins of a go routine which calls s.handleStream which invokes the method handler. 

in `stop()` 

```go
	for len(s.conns) != 0 {
		s.cv.Wait()
	}
```

^ this will ensure that all s.conns are removed before we stop 

```go
s.conns = nil 
```

^ this ensures we dont accept any new connections


```go
s.serveWG.Wait()
```

and then if we do this ^, we can make sure that all the goroutines created for every conn has completed.

Reply:
I'm not sure how would we handle the separate logic for Stop() and GracefulStop()

Reply:
I tried making this change to your fork and run tests. But they fail - possibly im missing something.

I'm approving this since this change fixes the issue.

Reply:
I think it could possibly work for GracefulStop, but since we need Stop to wait for the connections to close but not for the handlers to return, I don't think it's possible.  Otherwise, that would be a good idea.